src/sonic-config-engine/minigraph.py

```diff
@@ -1002,14 +1008,28 @@ def parse_linkmeta(meta, hname):
                 linkmetas[port]["PeerSwitch"] = upper_tor_hostname
         if auto_negotiation:
             linkmetas[port]["AutoNegotiation"] = auto_negotiation
+        if macsec_enabled:
+            linkmetas[port]["MacSecEnabled"] = macsec_enabled
     return linkmetas
 
+def parse_macsec_profile(val_string):
+    macsec_profile = {}
+    values = val_string.strip().split(' ')
```

Comment:
>val_string [](http://example.com/codeflow?start=13&length=10)

The parsing logic is too strict and not easy to read/extend. Could you explore libraries like [https://docs.python.org/3/library/configparser.html](https://docs.python.org/3/library/configparser.html)?

Reply:
@Qi I am finding this config parser library used when we have to read from an input file and while parsing it has restrictions like need to have [SECTION] header etc - which I feel is not applicable here.

I checked the following, and the current parsing should take care of them.
> the macsec profile name cannot be with spaces
> the profile will have PrimaryKey and FallbackKey.